src/main/java/org/opensearch/neuralsearch/search/collector/PagingFieldCollector.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.search.collector;
+
+import java.io.IOException;
+import java.util.Objects;
+import org.apache.lucene.index.LeafReaderContext;
+import org.apache.lucene.search.FieldDoc;
+import org.apache.lucene.search.LeafCollector;
+import org.apache.lucene.search.Sort;
+import org.opensearch.common.Nullable;
+import org.opensearch.neuralsearch.search.HitsThresholdChecker;
+
+/**
+ * PagingFieldCollector collects the sorted results at the shard level for every individual query
+ * as per search_after criteria applied in the search request.
+ * It collects the list of TopFieldDocs.
+ */
```

Comment:
@vibrantvarun Given hybrid query doesn't support pagination and I know we want to build this pagination with sort, my worry is customer behavior will be broken where with normal hybrid search pagination doesn't work with then moment we add sort it starts to work. What is your thought?

Reply:
Let me put it this way,

Pagination is of 4 types
1) from and size parameters
2) Scroll
3) `search_after` exclusively used with sorting
4) Point in time with `search_after`

Here I have covered only 3rd point that just not as pagination but an extended use case for sort. The reason why I have covered it is because it is almost the similar amount of work. For customer, It will be like `search_after` with sorting. Moreover, the `search_after` is not used without sorting. So the customer will not try to use `search_after` with hybrid query without sort, therefore no  customer behavior would be there.

If `search_after` could be used with or without sorting then we can say there is a difference in customer behavior.

Reply:
As per our discussion and looking at this feature that search_after is used only when sorting is present, lets make sure that we document this properly in the docs. We should also mention in the docs that search_after is only supported with sorting. This part is crirtical